util/design/lib/LcStEnc.py

```diff
@@ -284,8 +290,18 @@ def __init__(self, config):
         log.info('Seed: {0:x}'.format(config['seed']))
         log.info('')
 
-        # Re-initialize with seed to make results reproducible.
-        random.seed(LC_SEED_DIVERSIFIER + int(config['seed']))
+        if config['entropy_buffer']:
+            # Load entropy from a file
+            strong_random.load(config['entropy_buffer'])
+        else:
+            # Re-initialize with seed to make results reproducible.
+            # Generate entropy buffer from the seed.
+            ebg.gen_buffer(ENTROPY_BUFFER_SIZE_BYTES, PATH_TO_ENTROPY_BUFFER, False,
```

Comment:
I don't think this is why this is failing CI, but one potential additional issue I could think of is that some DV sim tests in chip_sim_cfg.hjson (namely: `chip_sw_power_virus` and `chip_sw_flash_scrambling_smoketest`) use bazel to pre-scramble the flash VMEM image for backdoor loading in sim. Since the scrambling keys are computed using netlist constants generated using this module, the `gen-flash-img.py` also invokes this module to get the same constants. However, it uses Bazel to invoke this python code, and since the `PATH_TO_ENTROPY_BUFFER` file is just a relative path encoded in the python module, Bazel will have no idea of its existence and it most likely will not get created in the Bazel sandbox. 

This issue is also present for all tests in chip_rom_tests.hjson that use Bazel to generate the OTP image (instead of the images generated [here](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/chip_sim_cfg.hjson;drc=05af19dc74e8a84750638d0ba71ffffaaa94c936;l=237) during the dvsim pre-build stage), namely any tests that use the `+use_otp_image=OtpTypeCustom`plusarg.

There are two ways to address this issue:
1. create a custom bazel rule that drives the scripts that consume this LcStateEnc module (and same with the OtpMemMap module) and ensure that we declare an output file this script will generate (i.e. the entropy buffer file); or
2. if the buffer file isn't present just fill the strong random buffer with entropy from the random module directly, instead of generating a file with entropy from random and reading in the file.

1 is ideal, but given that requires a bit of Bazel expertise, I think 2 is sufficient for now and I can refactor things later when we tie together things with Bazel. Does that sound good?

Again, I don't think this is the reason why tests are timing out in private CI, I think there is a different issue, but I think you will encounter what I describe above after you fix the current CI issue.